### PR TITLE
refactor: Migrate logging in `bot.py`

### LINKED ISSUE #1: Switch to structured logging
- [x] Find a Python logging framework that supports structured logging (ie, searchable data, not just a string blob)
- [ ] Move all the calls to `print()` to instead use the logging framework
- [ ] Possibly add more logging as needed

### PR DESCRIPTION
**Pull request by proegssilb**
Resolves #1 

Migrate all the debug `print()` calls to calls into `logging`.

### CODE REVIEW

**Comment by proegssilb** [comment]
I created this WIP PR so that people can look at what I'm up to in the rework.

File: bot.py [hunk 1]
```diff
@@ -52,11 +54,9 @@ async def format_times(times):
             if times2_str and (get_best_msg.part == 0 or get_best_msg.part == 2):
                 embed.add_field(name="Part 2", value=times2_str, inline=True)
             await msg.reply(embed=embed)
-        except Exception as e:
-            traceback.print_exception(
-                e
-            )  # TODO: we probably dont want to log all incorrectly formatteed messages long term
-            await msg.reply(f"Error handling your message", embed=constants.HELP_REPLY)
+        except Exception:
+            logger.exception("Errror while processing request for leaderboard.")
```
Comment (reticivis-net) [inline bot.py:58]: > Errror

lol

**Review by reticivis-net — APPROVED** [review]
looks fine except for One (1) typo

**Review by ultrabear — APPROVED** [review]
Aside from the typo, nothing looks wrong here

**Pull request merged**